scripts/tests/twister/test_quarantine.py

```diff
@@ -0,0 +1,195 @@
+        []
+    ),
+    (
+        ['all'],
+        ['dummy platform', 'another platform'],
+        ['dummy architecture', 'another architecture'],
+        ['dummy simulation', 'another simulation'],
+        None,
+        ['scenarios']
+    ),
+    (
+        ['dummy scenario', 'another scenario'],
+        ['dummy platform', 'all'],
+        ['all', 'another architecture'],
+        ['dummy simulation', 'another simulation'],
+        None,
+        ['platforms', 'architectures']
+    ),
+    (
+        ['all', 'another scenario'],
+        [],
+        [],
+        ['all', 'all'],
+        QuarantineException,
+        ['scenarios', 'platforms', 'architectures', 'simulations']
+    ),
+]
+
+
+@pytest.mark.parametrize("scenarios, platforms, architectures, simulations, expected_exception, empty_filter_attrs", TESTDATA_1)
+def test_quarantineelement_post_init(scenarios, platforms, architectures, simulations, expected_exception, empty_filter_attrs):
+    if expected_exception:
+        with pytest.raises(expected_exception):
+            quarantine_element = QuarantineElement(
+                scenarios=scenarios,
+                platforms=platforms,
+                architectures=architectures,
+                simulations=simulations
+            )
+    else:
+        quarantine_element = QuarantineElement(
+            scenarios=scenarios,
+            platforms=platforms,
+            architectures=architectures,
+            simulations=simulations
+        )
+
+        for attr in ['scenarios', 'platforms', 'architectures', 'simulations']:
+            if attr in empty_filter_attrs:
+                assert getattr(quarantine_element, attr) == []
+            else:
+                assert getattr(quarantine_element, attr) != []
+
+
+def test_quarantinedata_post_init():
+    quarantine_element_dict = {
+        'scenarios': ['all'],
+        'platforms': ['dummy platform'],
+        'architectures': [],
+        'simulations': ['dummy simulation', 'another simulation']
+    }
+
+    quarantine_element = QuarantineElement(
+        platforms=['dummy platform'],
+        architectures=[]
+    )
+    quarantine_element.scenarios = []
+    quarantine_element.simulations = ['dummy simulation', 'another simulation']
+
+    quarantine_data_qlist = [quarantine_element, quarantine_element_dict]
+
+    quarantine_data = QuarantineData(quarantine_data_qlist)
+
+    assert quarantine_data.qlist[0] == quarantine_data.qlist[1]
+
+
+TESTDATA_2 = [
+    (
+        '',
+        QuarantineData()
+    ),
+    (
+        '[\n {\n  "scenarios": ["all"],\n  "platforms": ["dummy platform"],\n  "architectures": [],\n  "simulations": ["dummy simulation", "another simulation"]\n }\n]',
+        QuarantineData([QuarantineElement(scenarios=[], platforms=['dummy platform'], architectures=[], simulations=['dummy simulation', 'another simulation'])])
+    ),
+    (
+        '[\n {\n  "I": ["am"],\n  "not": "a",\n  "valid": [],\n  "JSON": ["for", "this"]\n }\n]',
```

Comment:
you can try with import textwrap
```suggestion
        textwrap.dedent("""
        [
            {
                'I': ['am'],
                'not': 'a',
                'valid': [],
                'JSON': ['for', 'this']
            }
        ]
        """),
```